runners/flink/src/main/java/org/apache/beam/runners/flink/adapter/BeamFlinkDataSetAdapter.java

```diff
@@ -213,7 +216,8 @@ private <InputT> FlinkBatchPortablePipelineTranslator.PTransformTranslator flink
       // When we run into a FlinkInput operator, it "produces" the corresponding input as its
       // "computed result."
       String inputId = t.getTransform().getSpec().getPayload().toStringUtf8();
-      DataSet<InputT> flinkInput = Preconditions.checkNotNull( (DataSet<InputT>) inputMap.get(inputId));
+      DataSet<InputT> flinkInput =
+          Preconditions.checkNotNull((DataSet<InputT>) inputMap.get(inputId));
```

Comment:
Commented on other PR - use the stuff in org.apache.beam.sdk.util so that you can do `checkArgumentNotNull` or `checkStateNotNull` with an explicit message indicating the source of the problem.

Reply:
Good to keep in mind, but I'll push back a bit here and say there's not really a more meaningful error to give here (either to debug this or to communicate anything to the user). 

My philosophy for error messages is similar to that of comments, they should give information that is not present in the code. If they are simply redundant with what the code says then they just decrease the signal-to-noise ratio (as well as issues like having more potential to get out of date).

Reply:
I agree with the philosophy. Just see a different balance. The user is going to report the error, probably without stack trace, and (IMO correctly) view that NPE == segfault in terms of how it reflects on the quality of the software being built. So having a meaningful message and better exception accelerates communication and debugging a lot.

For example, some of these checks should say IllegalState (or IllegalArgument, debatable) with "corrupted portable pipeline proto" which is much clearer than having to read the code to determine it.

Reply:
OK, I'll buy the NPE == segfault argument (though this code will probably never ever get hit.) Updated to use our version.